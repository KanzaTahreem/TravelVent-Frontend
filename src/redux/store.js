import { configureStore } from '@reduxjs/toolkit';
import reserveItems from './reserveSlice';
// import cartItems from './cartSlice';
import tripSlice from './tripSlice';
import sessionSlice from './reducers/sessionSlice';

const store = configureStore({
  reducer: {
    reserveItems,
    trips: tripSlice,
    session: sessionSlice,
  },
});

export default store;
