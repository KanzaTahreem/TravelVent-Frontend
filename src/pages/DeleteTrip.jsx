import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FaTrashAlt } from 'react-icons/fa';
import { fetchTrips, deleteTrip } from '../redux/tripSlice';
import '../assets/stylesheets/delete.css';

const DeleteTrip = () => {
  const trips = useSelector((state) => state.trips.trips);
  const dispatch = useDispatch();
  const [deleteSuccess, setDeleteSuccess] = useState(false);
  const [deleteFailed, setDeleteFailed] = useState(false);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    dispatch(fetchTrips());
  }, [dispatch]);

  const handleDelete = async (id) => {
    setLoading(true);
    try {
      await dispatch(deleteTrip(id));
      setDeleteSuccess(true);
      setTimeout(() => {
        setDeleteSuccess(false);
      }, 3000);
    } catch (error) {
      setDeleteFailed(true);
      setTimeout(() => {
        setDeleteFailed(false);
      }, 3000);
    } finally {
      setLoading(false);
    }
  };

  const availableTrips = trips.length;

  const message = `We have ${availableTrips} trip${availableTrips !== 1 ? 's' : ''} available.
  ${availableTrips ? 'Select a trip from below to delete!' : 'Add some trips!'}`;

  return (
    <section className="delete-page">
      <div className="content">
        <h1>Available Trips</h1>
        <p className="message">{message}</p>
        {deleteSuccess && <p className="success-msg msg">Trip deleted successfully!</p>}
        {deleteFailed && <p className="error-msg msg">Failed to delete a trip!</p>}
        {loading && <p className="loading-msg msg">Deleting trip...</p>}
        <div className="all-trips">
          {trips.map((trip) => (
            <div className="single-trip" key={trip.id}>
              <img src={trip.image_url} alt={trip.destination_city} />
              <div className="trip-content">
                <h2>{trip.destination_city}</h2>
                <p>
                  $
                  {trip.price}
                </p>
                <button type="button" onClick={() => handleDelete(trip.id)} disabled={loading}>
                  <FaTrashAlt />
                  Delete
                </button>
              </div>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};

export default DeleteTrip;
