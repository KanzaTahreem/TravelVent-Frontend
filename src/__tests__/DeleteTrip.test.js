import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DeleteTrip from '../pages/DeleteTrip';

describe('RootLayout', () => {
  let deleteTrip;
  beforeAll(async () => {
    deleteTrip = renderer
      .create(
        <Provider store={store}>
          <Router>
            <DeleteTrip />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should renders correctly', () => {
    expect(deleteTrip).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(deleteTrip).toMatchSnapshot();
  });

  it('displays correct message when there are no trips', () => {
    const message = waitFor(() => screen.getByText(/Add some trips!/i));
    waitFor(() => expect(message).toBeInTheDocument());
  });
});
