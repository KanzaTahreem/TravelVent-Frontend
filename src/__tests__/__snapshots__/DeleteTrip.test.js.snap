// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should match the snapshot 1`] = `
<section
  className="delete-page"
>
  <div
    className="content"
  >
    <h1>
      Available Trips
    </h1>
    <p
      className="message"
    >
      We have 0 trips available.
  Add some trips!
    </p>
    <div
      className="all-trips"
    />
  </div>
</section>
`;
